packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-modal.tsx

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import { toPng } from 'html-to-image';
+import {
+    Button,
+    Clipboard,
+    DesktopWrapper,
+    Icon,
+    MobileWrapper,
+    Modal,
+    Text,
+    useCopyToClipboard,
+} from '@deriv/components';
+import { useIsMounted } from '@deriv/shared';
+import { observer } from '@deriv/stores';
+import { Localize, localize } from 'Components/i18next';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+import MyProfileSeparatorContainer from 'Components/my-profile/my-profile-separator-container';
+import { TAdvert } from 'Types';
+import ShareMyAdsIcons from './share-my-ads-socials';
+import ShareMyAdsCard from './share-my-ads-card';
+
+const ShareMyAdsModal = ({ advert }: TAdvert) => {
+    const [show_popup, setShowPopup] = React.useState(false);
+    const [is_copied, copyToClipboard, setIsCopied] = useCopyToClipboard();
+
+    const isMounted = useIsMounted();
+    const divRef = React.useRef(null);
+    // TODO: replace with proper url when available
+    const advert_url = window.location.href;
+
+    const { hideModal, is_modal_open } = useModalManagerContext();
+
+    let timeout_clipboard: ReturnType<typeof setTimeout>;
```

Comment:
```suggestion
    let timeout_clipboard: number;
```
setTimeout always return a number.

Reply:
man eslint will complain if i dont return it as typeof setTimeout

Reply:
its ok i removed this, and changed the way timeout was cleared